pulsar-broker-common/src/main/java/org/apache/pulsar/broker/authentication/AuthenticationState.java

```diff
@@ -0,0 +1,50 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.pulsar.broker.authentication;
+
+import java.io.IOException;
+import javax.naming.AuthenticationException;
+import org.apache.pulsar.common.api.AuthData;
+
+/**
+ * Interface for authentication state.
+ *
+ * It is basically holding the authentication state.
+ * It tell broker whether the authentication is completed or not,
+ * if completed, what is the AuthRole is.
+ */
+public interface AuthenticationState {
+    /**
+     * Get AuthenticationDataSource for this authentication state.
+     */
+    AuthenticationDataSource getAuthData();
+
+    /**
+     * After the authentication between client and broker completed,
+     * get authentication role represent for the client.
+     */
+    String getAuthRole() throws AuthenticationException;
+
+    /**
+     * Returns null if authentication has completed, and no auth data is required to send back to client.
+     * Do auth and Returns the auth data back to client, if authentication has not completed.
+     */
+    AuthData authenticate(AuthData authData) throws IOException;
```

Comment:
Why do you have this method here, and also a method to get the auth data source? You should have one or the other.

Reply:
Thanks @ivankelly , will remove getAuthDataSource, here want to do all the auth related work in authState.